messaginginapp/src/test/java/io/customer/messaginginapp/state/InAppMiddlewaresTest.kt

```diff
@@ -0,0 +1,98 @@
+package io.customer.messaginginapp.state
+
+import io.customer.messaginginapp.gist.data.model.Message
+import io.customer.messaginginapp.testutils.extension.testMatchesRoute
+import io.customer.sdk.core.di.SDKComponent
+import org.reduxkotlin.middleware
+
+/**
+ * Test-specific middleware for processing messages in the queue.
+ * This version uses our test-specific matching logic.
+ */
+internal fun testProcessMessages() = middleware<InAppMessagingState> { store, next, action ->
```

Comment:
Not sure I fully understand the purpose of these test middlewares and how they're used. Can you please add a bit more context to make it easier for others to understand when working with these in future?